india_forecast_app/app.py

```diff
@@ -1,10 +1,140 @@
+import datetime as dt
+import logging
+
 import click
 
+from .model import DummyModel
+
+log = logging.getLogger(__name__)
+
+
+def _get_site_ids() -> list[str]:
+    """
+    Gets all avaiable site_ids in India
+
+    Returns:
+            A list of site_ids
+    """
+
+    return [
+        "b0579f31-70d9-4682-962e-4e2b30fa1e85",
+        "d0146492-90d2-41bf-9e44-153032492bad",
+    ]
+
+
+def _get_model():
+    """
+    Instantiates and returns the forecast model ready for running inference
+
+    Returns:
+            A forecasting model
+    """
+
+    model = DummyModel()
+    return model
+
+
+def _run_model(model, site_id: str, timestamp: dt.datetime):
+    """
+    Runs inference on model for the given site & timestamp
+
+    Args:
+            model: A forecasting model
+            site_id: A specific site ID
+            timestamp: timestamp to run a forecast for
+
+    Returns:
+            A forecast or None if model inference fails
+    """
+
+    try:
+        forecast = model.predict(site_id=site_id, timestamp=timestamp)
+    except Exception:
+        log.error(
+            f"Error while running model.predict for site_id={site_id}. Skipping",
+            exc_info=True,
+        )
+        return None
+
+    return forecast
+
+
+def _save_forecast(site_id: str, timestamp: dt.datetime, forecast, write_to_db: bool):
+    """
+    Saves a forecast for a given site & timestamp
+
+    Args:
+            site_id: A specific site ID
+            timestamp: timestamp to run a forecast for
+            forecast: a forecast containing predicted generation values for the given site
+            write_to_db: If true, forecast values are written to db, otherwise to stdout
+
+    Raises:
+            IOError: An error if database save fails
+    """
+
+    if write_to_db:
+        pass
```

Comment:
we can use https://github.com/openclimatefix/pv-site-datamodel/blob/main/pvsite_datamodel/write/generation.py#L30 to save the forecast

Would need to set up session using something like [this](https://github.com/openclimatefix/pv-site-production/blob/main/forecast-inference/forecast_inference/app.py#L190) and then get session using [this](https://github.com/openclimatefix/pv-site-production/blob/main/forecast-inference/forecast_inference/app.py#L67)

Reply:
The forecast app should be writing forecast values though, not generation values, no?

I think I'd need to add something in to the [pvsite-datamodel](https://github.com/openclimatefix/pv-site-datamodel) repo to save forecasts (in a similar fashion to how generation values are written in https://github.com/openclimatefix/pv-site-datamodel/blob/main/pvsite_datamodel/write/generation.py#L30

Reply:
Yea, yes please add this to the datamodel